### PR TITLE
Add "standard" Axes wrapper getters/setters for Axis invertedness.

### PR DESCRIPTION
**Pull request by anntzer**
Currently, toggling an axis invertedness requires either using the Axes method `ax.invert_xaxis()` (whose effect depends on whether it has already been called before) or going through the axis method `ax.xaxis.set_inverted()`; likewise, querying invertedness state requires either using the `ax.xaxis_inverted()` method, which has slightly nonstandard naming, or going through the axis method `ax.xaxis.get_inverted()`.

For practicality, provide getters and setters with standard names: `ax.get_xinverted()`/`ax.set_xinverted()` (and likewise for y/z).  In particular, the "standard" setter can be used with the multi-setter Artist.set(), or directly when creating the axes
(`add_subplot(xinverted=True)`).

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by timhoffm** [comment]
My 2 cents:

- Generally, I advocate to not replicate all Axis methods on the Axes. We intentionally made the properties `xaxis` / `yaxis` available to have an alomost as short access path `ax.xaxis.get_inverted()` vs. `ax.get_xinverted()`. So if inversion was not yet available on Axes level, I would not add it.
- The existing functions are `invert_xaxis()` / `xaxis_inverted()` are unfortunate. They do not follow the standard get/set naming patterns and it's not clear (neither from docstring nor from documentation that `invert_xaxis()` is a toggle, rather than setting the inversion unconditionally.
- 

In the spirit of API consistency, we should move away from the old functions. This means at least discouraging. I'm unsure on deprecation. As an replacement, I'd be fine with having inversion only available on the axis and thus reduce the Axes API footprint.  This would mean you don't get the standard axes property benefits like `add_subplot(xinverted=True)`- you don't have them currently, so it's at least not worsening the API. But if you feel strongly, that we should pull this up into the Axes API, I'd be ok with it as well.

**Comment by anntzer** [comment]
FWIW I do `ax = plt.figure(...).add_subplot(..., aspect=1); ax.yaxis.set(inverted=True)` *very* often (for plotting things (e.g. particle trajectories) where coordinates would match image coordinates but without having an imshow() call that would implicitly do the `set_aspect(1); invert_yaxis()` dance for me, and this is why I propose to make `add_subplot(..., aspect=1, yinverted=True)` possible.

(Yes, I could write `ax.invert_yaxis()` instead of `ax.yaxis.set(inverted=True)` but I don't like it for the reasons already discussed.)

**Comment by jklymak** [comment]
I was going to say that images invert the axis all the time.  Also in my field, we plot depth on an inverted axis all the time.  Just to say that I don't think inverting an axis is very rare.  I think the proposed changes here are straight forward enough that they are OK from an API point of view - I agree that its not nice to have many ways to do the same thing, but in this case "the same thing" is pretty straight forward and well-defined.

**Comment by timhoffm** [comment]
Semi-OT: Reason why `ax.invert_xaxis()` is odd: By itself this is a valid name (a combination verb_noun), and we have legitimate other names with that pattern, e.g. `Axes.update_datalim`. The point here is though, that we don't regard inversion of an axis an operation; it's rather a state ("is the axis inverted or not"). While one can descripe every change of state as an operation (inverting results in an inverted state), states are more specific in that they can also be queried. Therefore the state associated set/get methods are more appropriate than an operation method.

Note: `invert_xaxis()` is the only case of this kind of pattern in Axes. Getting rid of it makes the API more consistent. For reference, there are other oddballs (not suggesting to do something about it):
- `margins()`, `grid()`, `tick_params()`, `locator_params()` - these also set state and should be `set_*`methods.
- `xaxis_date()` / `yaxis_date()` - these are weirdly specific

**Comment by timhoffm** [comment]
@anntzer can you please discourage use of the old syntax?

- add
  ```
        .. admonition:: Discouraged
        
           The use of this method is discouraged.
           Use `.Axes.set_xinverted` instead.
  ```
  to the old functions
- mention the preference for the new functions in the what's new entry.

**Comment by rcomer** [comment]
An inverted y-axis is also common in atmospheric science, when we plot against pressure.

**Comment by anntzer** [comment]
Edited as suggested by @timhoffm.
FWIW I suspect that the name of tick_params() / locator_params() comes from the coresponding pyplot functions, which were then directly transposed as axes methods.  Note that there is a corresponding set_tick_params() axis method (but no set_locator_params() likely because one can easily directly access the locator itself and call set_params() on it).

**Comment by timhoffm** [comment]
Logically 👍 

You need to fix the linting errors. And I don't know what to make of the failing tests. Somehow 3d stuff seems broken.

**Comment by timhoffm** [comment]
Doc build failures are real (reference issues related to the PR).

**Review by timhoffm — APPROVED** [review]

File: lib/matplotlib/axes/_base.py [hunk 1]
```diff
@@ -3603,15 +3604,30 @@ def invert_xaxis(self):
...
+        get_xinverted
         get_xlim, set_xlim
         get_xbound, set_xbound
         """
         self.xaxis.set_inverted(not self.xaxis.get_inverted())
 
+    set_xinverted = _axis_method_wrapper("xaxis", "set_inverted")
+    get_xinverted = _axis_method_wrapper("xaxis", "get_inverted")
     xaxis_inverted = _axis_method_wrapper("xaxis", "get_inverted")
+    if xaxis_inverted.__doc__:
```
Comment (jklymak) [inline lib/matplotlib/axes/_base.py:3623]: Why is this necessary?

Comment (anntzer) [reply lib/matplotlib/axes/_base.py:3623]: `__doc__` can be unset in "optimized" mode (https://docs.python.org/3/using/cmdline.html#cmdoption-OO).  There's already a few other places in the codebase with the same check.

Comment (jklymak) [reply lib/matplotlib/axes/_base.py:3623]: Sorry, still not following what this does or why...  Is this because we are building `__doc__` dynamically?

Comment (timhoffm) [reply lib/matplotlib/axes/_base.py:3623]: Yes, in optimized mode the following line `xaxis_inverted.__doc__ += …” would raise and therefore must be guarded.

**Comment by openlitchi** [comment]
Hello, I noticed in the [Highlights](https://github.com/matplotlib/matplotlib/releases/tag/v3.10.0) of v3.10.0 it mentions **_Standard getters/setters for axis inversion state_**. However, on my Windows machine, I am actually unable to access the **matplotlib.axes.Axes.get_xinverted** function in `matplotlib 3.10.0`, even though this function is present in the [web documentation](https://matplotlib.org/devdocs/api/_as_gen/matplotlib.axes.Axes.get_xinverted.html) for the dev version. I could only find this related Pull Request. Could it be that this PR was not merged into v3.10.0?

image

**Comment by QuLogic** [comment]
@ksunden that should not be in the release notes; this change landed far too late for 3.10, and wasn't backported.
